### PR TITLE
Fix an issue with the default value of 'Checkbox' and 'Toggle' fields

### PR DESCRIPTION
**Pull request by kienstra**
#### Background
* I introduced a bug into the newly released editor where the default of the Toggle and Checkbox fields is truthy instead of falsey
* The data type of the default was also wrong, though it was wrong before (also my fault 😂 ). It should be a `boolean`, but it was stored as `0` or `1` 

#### Changes
* Makes the default for Toggle and Checkbox fields `false`, how it used to be in GCB `1.0.3`: here-false

* Changes the data type of the default value from a string like `'0'` or `'1'` to a `boolean`
* But most of this diff is deleting dead code from the previous editor that I forgot to delete. It's slightly related to this fix, as fields have `->sanitize` and `->validate` functions that aren't used anymore and confusing.
* To test with a previously-created field, you might have to go to: new editor  > Toggle field > Default > check and uncheck, then click 'Update: 
testing-up

#### Testing instructions
1. `composer install && npm install && npm run dev`
2. `/wp-admin` > Custom Blocks > Add New
3. Name the block 'Test Checkbox'
4. Add a field
5. Give it any 'Field Label'
6. Change its type to 'Checkbox': expected-unchecked
7. Publish the block
8. Create a new post
9. Add the 'Test Checkbox' block you created: 
unchecked-here
10. Repeat steps 2-8, except with a 'Toggle' field type 
11. Expected: it's untoggled by default: 
untoggled-by-default

### CODE REVIEW

File: js/src/block-editor/controls/checkbox.js [hunk 1]
```diff
@@ -15,7 +20,6 @@ const GcbCheckboxControl = ( props ) => {
 			label={ field.label }
 			help={ field.help }
 			checked={ value }
-			options={ field.options }
```
Comment (kienstra) [inline js/src/block-editor/controls/checkbox.js:18]: It looks like this is no longer a property of `<CheckboxControl>`

File: php/Blocks/Controls/ControlAbstract.php [hunk 2]
```diff
@@ -85,39 +85,34 @@ public function __construct() {
 	public function create_settings_config() {
 		$this->settings_config = [
 			'location'    => [
-				'name'     => 'location',
-				'label'    => __( 'Field Location', 'genesis-custom-blocks' ),
-				'type'     => 'location',
-				'default'  => 'editor',
-				'sanitize' => [ $this, 'sanitize_location' ],
```
Comment (kienstra) [inline php/Blocks/Controls/ControlAbstract.php:92]: The `sanitize` key was from the PHP editor, where it sanitized the new values server-side.

File: php/Blocks/Controls/ControlAbstract.php [hunk 3]
```diff
@@ -153,436 +148,4 @@ static function( $setting ) {
...
-
-	/**
-	 * Render additional settings in table rows.
-	 *
-	 * @param Field  $field The Field containing the options to render.
-	 * @param string $uid   A unique ID to used to unify the HTML name, for, and id attributes.
-	 *
-	 * @return void
-	 */
-	public function render_settings( $field, $uid ) {
```
Comment (kienstra) [inline php/Blocks/Controls/ControlAbstract.php:165]: I should have deleted this earlier. It's from the previous editor, when it was PHP-rendered.

File: js/src/edit-block/components/settings/checkbox.js [hunk 4]
```diff
@@ -28,7 +29,6 @@ const Checkbox = ( { handleOnChange, setting, value } ) => {
 				id={ id }
 				className="ml-2"
 				type="checkbox"
-				value={ checked }
```
Comment (kienstra) [inline js/src/edit-block/components/settings/checkbox.js:31]: It [looks like](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox) `value` isn't required

File: php/Blocks/Controls/Checkbox.php [hunk 5]
```diff
@@ -49,11 +49,10 @@ public function register_settings() {
...
 			[
-				'name'     => 'default',
-				'label'    => __( 'Default Value', 'genesis-custom-blocks' ),
-				'type'     => 'checkbox',
-				'default'  => '0',
-				'sanitize' => [ $this, 'sanitize_checkbox' ],
+				'name'    => 'default',
+				'label'   => __( 'Default Value', 'genesis-custom-blocks' ),
+				'type'    => 'checkbox',
+				'default' => false,
```
Comment (kienstra) [inline php/Blocks/Controls/Checkbox.php:55]: The default should be a `boolean`, as a `Checkbox` field has a type of `boolean`

File: php/Blocks/Controls/ControlAbstract.php [hunk 6]
```diff
@@ -141,448 +136,7 @@ abstract public function register_settings();
 	 */
 	public function jsonSerialize() {
 		$object = clone( $this );
-
 		unset( $object->settings_config );
-		$object->settings = array_map(
-			static function( $setting ) {
-				unset( $setting->sanitize, $setting->validate );
```
Comment (kienstra) [inline php/Blocks/Controls/ControlAbstract.php:148]: These properties no longer exist, so no need to unset them

File: php/Blocks/Controls/ControlAbstract.php [hunk 7]
```diff
@@ -141,448 +136,7 @@ abstract public function register_settings();
...
-	/**
-	 * Validate that the value is contained within a list of options,
-	 * and if not, return the first option.
-	 *
-	 * @param mixed $value    The value to be validated.
-	 * @param array $settings The field settings.
-	 *
-	 * @return mixed
-	 */
-	public function validate_options( $value, $settings ) {
```
Comment (kienstra) [inline php/Blocks/Controls/ControlAbstract.php:552]: This is now done in JavaScript

File: php/Blocks/Controls/Textarea.php [hunk 8]
```diff
@@ -50,86 +50,36 @@ public function register_settings() {
...
-	/**
-	 * Renders a <select> of new line rendering formats.
-	 *
-	 * @param ControlSetting $setting The ControlSetting being rendered.
-	 * @param string         $name    The name attribute of the option.
-	 * @param string         $id      The id attribute of the option.
-	 *
-	 * @return void
-	 */
-	public function render_settings_new_line_format( $setting, $name, $id ) {
```
Comment (kienstra) [inline php/Blocks/Controls/Textarea.php:99]: This is done in JS in the new editor

File: js/src/block-editor/controls/checkbox.js [hunk 9]
```diff
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
```
Comment (kienstra) [inline js/src/block-editor/controls/checkbox.js:4]: So TypeScript type checking works

Comment (kienstra) [reply js/src/block-editor/controls/checkbox.js:4]: (That showed the bug below where the `options` prop doesn't exist anymore)

File: js/src/edit-block/components/field-settings.js [hunk 10]
```diff
@@ -58,7 +58,7 @@ const FieldSettings = ( {
 
 					const SettingComponent = getSettingsComponent( setting.type );
 					const key = `field-setting-${ index }`;
-					const value = null === field[ setting.name ] ? setting.default : field[ setting.name ];
+					const value = field.hasOwnProperty( setting.name ) ? field[ setting.name ] : setting.default;
```
Comment (kienstra) [inline js/src/edit-block/components/field-settings.js:61]: `null === field[ setting.name ]` was the wrong logic. 

It should have been `undefined === field[ setting.name ]`, but  I think `field.hasOwnProperty( setting.name ) ` is even simpler.

File: php/Blocks/Controls/ControlSetting.php [hunk 11]
```diff
@@ -49,20 +49,6 @@ class ControlSetting {
...
-	 * @var mixed
-	 */
-	public $sanitize = '';
-
-	/**
-	 * Validating function.
-	 *
-	 * @var mixed
-	 */
-	public $validate = '';
```
Comment (kienstra) [inline php/Blocks/Controls/ControlSetting.php:64]: Remnants of the PHP editor

File: js/src/block-editor/controls/toggle.js [hunk 12]
```diff
@@ -1,20 +1,25 @@
...
 /**
  * WordPress dependencies
  */
 import { ToggleControl } from '@wordpress/components';
 
 const GcbToggleControl = ( props ) => {
 	const { field, onChange, getValue } = props;
-	const attr = { ...props.attributes };
-	if ( 'undefined' === typeof attr[ field.name ] ) {
-		attr[ field.name ] = field.default;
```
Comment (kienstra) [inline js/src/block-editor/controls/toggle.js:10]: This made `field.default` useless, as `attr` was never used again in the component.

File: js/src/block-editor/controls/toggle.js [hunk 13]
```diff
@@ -1,20 +1,23 @@
...
-	}
+	const initialValue = getValue( props );
+	const value = 'undefined' !== typeof initialValue ? initialValue : field.default;
```
Comment (kienstra) [inline js/src/block-editor/controls/toggle.js:14]: Similar to how other control components get the `value`

**Comment by kienstra** [comment]
Hi @RobStino,
Could you please test this when you can? Ideally it'll fix the issue you found.

Thanks!

**Review by dreamwhisper — APPROVED** [review]
This is working as described for me!

**Comment by kienstra** [comment]
Thanks so much, Jen!

**Comment by kienstra** [comment]
Hi @RobStino,
Could you please review this when you can?

It's supposed to fix the bug you found.

Thanks!

**Review by RobStino — APPROVED** [review]
Hi @kienstra 
Tested this and confirmed that the default of new toggle and checkbox fields was set to false in both the block builder and when adding the block to a post or page.
Thanks for the fix!

**Comment by kienstra** [comment]
Thanks, @RobStino!
Eeeeeeeeh!

**Comment by kienstra** [comment]
Here's a `.zip` of this branch: 
[genesis-custom-blocks.1.1.0.zip](https://github.com/studiopress/genesis-custom-blocks/files/6166890/genesis-custom-blocks.1.1.0.zip)

**Pull request merged**